lib/task/import/validate/csvBaseValidator.class.php

```diff
@@ -52,6 +54,41 @@ public function __construct(?array $options = null)
         $this->testData = new CsvValidatorResult($this->title, $this->filename, $this->displayFilename, $this->getClassName());
     }
 
+    public function columnPresent(string $columnName, ?array $header = null)
+    {
+        if (!isset($this->columnPresent[$columnName]) && null !== $header) {
+            $this->columnPresent[$columnName] = in_array($columnName, $header);
+        }
+
+        if (isset($this->columnPresent[$columnName])) {
+            return $this->columnPresent[$columnName];
+        }
+
+        return false;
+    }
+
+    public function columnDuplicated(string $columnName, ?array $header = null)
+    {
+        if ($this->columnPresent($columnName, $header)) {
```

Comment:
Opinion: To avoid the nested conditionals I'd prefer:

```
if (!$this->columnPresent($columnName, $header)) {
    return false;
}
```

Reply:
👍